File: 프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.java
class Solution {
    public int solution(int[] dot) {
        int answer = 0;
        int x = dot[0];
        int y = dot[1];
        
        if(x > 0) {
            answer = y > 0 ? 1 : 4;
        }else {
            answer = y > 0 ? 2 : 3; 
        }
        
        return answer;
    }
}

// 다른 풀이
int xParam = dot[0] < 0 ? 1 : 0;
int yParam = dot[1] < 0 ? 2 : 0;

return 1 + ((xParam | yParam) ^ (yParam >> 1));
// 비트 연산을 이용한 풀이
// (xParam | yParam) : 두 변수 중 하나라도 해당 비트가 설정되어 있으면 결과 설정 
// -- xParam이 1이고, yParam이 2라면 3 반환
// (yParam >> 1) : yParam을 오른쪽으로 1비트 시프트한 값
// -- yParam이 2일 경우 1 반환
// ^ : XOR 연산, 좌 우측 연산의 값이 다른 경우에만 1 반환, 같으면 0 반환
// 1 + : 마지막으로 결과에 1을 더함

File: 프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.py
def solution(dot):
    answer = 0
    
    if dot[0] > 0:
        if dot[1] > 0:
            answer = 1
        else:
            answer = 4;
    elif dot[0] < 0:
        if dot[1] > 0:
            answer = 2;
        else:
            answer = 3;
    
    return answer

# 다른 풀이
quad = [(3,2),(4,1)]
return quad[dot[0] > 0][dot[1] > 0]
# 1. quad라는 이름의 리스트 정의
# 2-1. quad[dot[0] > 0] : dot[0]이 0보다 크면 True, 아니면 False
# 2-2. quad[dot[1] > 0] : dot[1]이 0보다 크면 True, 아니면 False
# 3. quad 리스트에서 선택된 요소 반환
    # - 예시) dot = [2, 3]일 경우   -> quad[1][1] -> 1 사분면
    # - 예시) dot = [-2, 3]일 경우  -> quad[0][1] -> 2 사분면
    # - 예시) dot = [-2, -2]일 경우 -> quad[0][0] -> 3 사분면
    # - 예시) dot = [2, -3]일 경우  -> quad[1][0] -> 4 사분면

# 다른 풀이
a, b = 1, 0
if dot[0]*dot[1] > 0:
    b = 1
if dot[1] < 0:
    a = 2
return 2*a-b

# 다른 풀이
x , y = dot
if x*y > 0:
    return 1 if x>0 else 3
else:
    return 4 if x>0 else 2
